sdk/core/core/src/az_http_policy.c

```diff
@@ -56,14 +62,104 @@ AZ_NODISCARD az_result az_http_pipeline_policy_retry(
   return az_http_pipeline_nextpolicy(p_policies, hrb, response);
 }
 
+enum { AZ_HTTP_POLICY_AUTH_BUFFER_SIZE = 2 * 1024 };
+
+static AZ_NODISCARD az_result
+az_auth_get_resource_url(az_span const request_url, az_span_builder * p_builder) {
+  size_t scheme_size = 0;
+
+  uint8_t const * url = request_url.begin;
+  size_t const url_size = request_url.size;
+  for (size_t i = 0; i < url_size; ++i) {
+    if (url[i] == ':' && (i < url_size + 2) && url[i + 1] == '/' && url[i + 2] == '/') {
```

Comment:
What happen if `i + 1` and `i + 2` is bigger than request_url.size?

Reply:
impossible.